src/waveform/renderers/allshader/waveformrendererpreroll.cpp

```diff
@@ -12,6 +12,53 @@
 #include "widget/wskincolor.h"
 #include "widget/wwidget.h"
 
+namespace {
+QOpenGLTexture* generateTexture(float markerLength,
```

Comment:
I'm sorry for expanding the scope yet again, but this needs a little more supporting refactoring. The pointer returned here (and thus by createTexture) is owning. We should really avoid those in new code. Can you refactor this and `createTexture` to return by value or `unique_ptr` (preferably the former).

Reply:
sure. i will use a unique_ptr though

Reply:
Sure... seems a little unnecessary, but I won't insist.